Extension/src/LanguageServer/Providers/foldingRangeProvider.ts

```diff
@@ -21,36 +21,34 @@ export class FoldingRangeProvider implements vscode.FoldingRangeProvider {
             uri: document.uri.toString()
         };
         return new Promise<vscode.FoldingRange[]>((resolve, reject) => {
```

Comment:
I think you should be able to convert provideFoldingRanges() (and any other VS Code API that returns a ProviderResult) into an async function.  If so, you could remove this creation of a Promise.